pytorch_tabular/models/base_model.py

```diff
@@ -159,38 +169,69 @@ def calculate_metrics(self, y, y_hat, tag):
                 prog_bar=True,
             )
         return metrics
-    
+
     def data_aware_initialization(self, datamodule):
         pass
 
-    @abstractmethod
+    def compute_backbone(self, x: Dict):
+        # Returns output
+        x = self.backbone(x)
+        return x
+
+    def compute_head(self, x: Tensor):
+        y_hat = self.head(x)
+        if (self.hparams.task == "regression") and (
+            self.hparams.target_range is not None
+        ):
+            for i in range(self.hparams.output_dim):
+                y_min, y_max = self.hparams.target_range[i]
+                y_hat[:, i] = y_min + nn.Sigmoid()(y_hat[:, i]) * (y_max - y_min)
+        if type(self.head) == nn.Identity:
```

Comment:
This condition for checking nn.Identity is to distinguish SSL workflow from regular? If yes, can we add a comment there?

Reply:
Or use the same hparams.task==ssl condition

Reply:
I have add a comment, the Identity is used for Tabnet, this means that we cannot divide the model in backbone and head, so we cannot extract backbone features